src/when-contexts.ts

```diff
@@ -0,0 +1,29 @@
+import * as vscode from 'vscode';
+import { deepEqual } from './util/object'
+import * as docMirror from './doc-mirror'
+import * as context from './cursor-doc/cursor-context';
+
+let lastContexts: context.CursorContext[] = [];
+
+export default function setCursorContextIfChanged(editor: vscode.TextEditor) {
+    if (!editor || !editor.document || editor.document.languageId !== 'clojure') return;
+
+    const currentContexts = determineCursorContexts(editor.document, editor.selection.active);
+    setCursorContexts(currentContexts);
+}
+
+function determineCursorContexts(document: vscode.TextDocument, position: vscode.Position): context.CursorContext[] {
+    const mirrorDoc = docMirror.getDocument(document);
+    return context.determineContexts(mirrorDoc);
+}
+
+function setCursorContexts(currentContexts: context.CursorContext[]) {
+    if (deepEqual(lastContexts, currentContexts)) {
```

Comment:
I would expect this check to be in `setCursorContextIfChanged` above, and then this function to only be called if last and current are not equal. Then this function doesn't worry about that part, and the `if` version does (as its name describes).

Reply:
Good catch!